.github/workflows/test.yml

```diff
@@ -34,30 +34,40 @@ jobs:
       - name: Test
         run: python -m pytest -v -m "not ctest"
 
-  build-openassetio:
-    name: Build OpenAssetIO
-    runs-on: ubuntu-latest
-    container:
-      image: ghcr.io/openassetio/openassetio-build
-    steps:
-      - uses: actions/checkout@v3
-      - name: Build
-        uses: ./.github/build_openassetio
-
   ctest:
-    name: C++ tests
-    runs-on: ubuntu-latest
-    needs: build-openassetio
-    container:
-      image: aswf/ci-vfxall:2022-clang14.3
+    name: C++ tests ${{ matrix.config.os }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - os: windows-2019
+            site-packages: Lib/site-packages
+          - os: ubuntu-20.04
+            site-packages: lib/python3.9/site-packages
+          - os: macos-11
+            site-packages: lib/python3.9/site-packages
+    defaults:
+      run:
+        shell: bash
     steps:
       - uses: actions/checkout@v3
 
-      - name: Get OpenAssetIO
-        uses: actions/download-artifact@v3
+      - uses: actions/setup-python@v4
         with:
-          name: OpenAssetIO Build
-          path: ${{ github.workspace }}/openassetio
+          python-version: 3.9
+
+      - name: Download Latest OpenAssetIO Artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          name: openassetio-${{ matrix.config.os }}
+          repo: openassetio/openassetio
+          workflow: build.yml
+          branch: main
+          path: openassetio
+
+      - name: Set CMAKE_PREFIX_PATH
+        run: echo "CMAKE_PREFIX_PATH=${{ github.workspace }}/openassetio" >> $GITHUB_ENV
```

Comment:
[consider][strong] Did I understand right that the reason for doing it this way vs `env` was for consistency with other CI that has multiple steps? If so, then feels like either `PYTHONPATH` should be done here too, or just pop it back in `env` in the test step for ease of understanding when youre skim reading and trying to work out what the env for that step is. I think I'd personally prefer the latter as its a little more "vanilla", and we're not reusing it here.

Reply:
Since both yourself and Davids intuition was to have a direct environment setup in each step, deferred to your intuition : https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/56/commits/84f8486450baa1ff5546d9217e2838a870fc31c4

Reply:
Thanks - doesn't mean that trick can't be used else where:-)